### PR TITLE
Improve/clean-up camel-quarkus-validator extension tests.

### PR DESCRIPTION
**Pull request by svkcemk**

### CODE REVIEW

File: integration-tests/validator/src/main/java/org/apache/camel/quarkus/component/validator/it/ValidatorResource.java
Comment (aldettinger) [inline integration-tests/validator/src/main/java/org/apache/camel/quarkus/component/validator/it/ValidatorResource.java:40]: `processOrderInXmlFromFileSystem` => Don't we do more than file system ?

Comment (svkcemk) [reply integration-tests/validator/src/main/java/org/apache/camel/quarkus/component/validator/it/ValidatorResource.java]: Thanks just renamed to a generic one.

**Comment by aldettinger** [comment]
@svkcemk It's look good to me. Well done :+1: 

Just one more thing. It's really nice to have separate commits during review. Before merging, it would be good to rebase all commits into a single one in order to help maintaining a clear git history.

**Comment by svkcemk** [comment]
Thanks @aldettinger I was waiting for your review, now commits are squashed into one single commit.

**Review by aldettinger — APPROVED** [review]
Looks good to me :+1:  Let's merge after ci build for the sake of completeness.

File: integration-tests/validator/src/main/java/org/apache/camel/quarkus/component/validator/it/ValidatorRouteBuilder.java
Comment (ppalaga) [inline integration-tests/validator/src/main/java/org/apache/camel/quarkus/component/validator/it/ValidatorRouteBuilder.java:53]:
```suggestion
        try (InputStream resourceAsStream = Thread.currentThread().getContextClassLoader().getResourceAsStream(sourceXsd)) {
```

Otherwise the stream is not closed. It may cause problems especially on Windows.

File: integration-tests/validator/src/main/java/org/apache/camel/quarkus/component/validator/it/ValidatorRouteBuilder.java
Comment (ppalaga) [inline integration-tests/validator/src/main/java/org/apache/camel/quarkus/component/validator/it/ValidatorRouteBuilder.java:58]:
```suggestion
            throw new RuntimeException("Could not read "+ sourceXsd + " from classpath or copy it to " + tempXsd, e);
```

The wrapped exception message does not say related to which file the operation failed. Adding the path to the wrapper exception will help future users to faster figure out where the problem is.

File: integration-tests/validator/src/test/java/org/apache/camel/quarkus/component/validator/it/ValidatorTest.java
Comment (ppalaga) [inline integration-tests/validator/src/test/java/org/apache/camel/quarkus/component/validator/it/ValidatorTest.java:52]: Here I wonder whether there is some body text (or any other way) from which we could see that what failed is really the XSD validation? 500 could mean a failure somewhere else in the stack, which we definitely would not want to accept as "test passed".

**Review by ppalaga — COMMENTED** [review]
Very nice, thanks @svkcemk ! A few small nitpicks inline.

File: integration-tests/validator/src/main/java/org/apache/camel/quarkus/component/validator/it/ValidatorRouteBuilder.java
Comment (svkcemk) [reply integration-tests/validator/src/main/java/org/apache/camel/quarkus/component/validator/it/ValidatorRouteBuilder.java]: Thanks @ppalaga added some changes.

File: integration-tests/validator/src/test/java/org/apache/camel/quarkus/component/validator/it/ValidatorTest.java
Comment (svkcemk) [reply integration-tests/validator/src/test/java/org/apache/camel/quarkus/component/validator/it/ValidatorTest.java]: @ppalaga I have added some checks based on the body, I think it will make more sense now, Thanks for pointing out !

Comment (ppalaga) [reply integration-tests/validator/src/test/java/org/apache/camel/quarkus/component/validator/it/ValidatorTest.java]: The tests are failing. You might want to forward the exception message to the client in `ValidatorResource`. Something like 

```
    public javax.ws.rs.core.Response processOrderInXml(String statement, @PathParam("scheme") String scheme) {
        try {
            return javax.ws.rs.core.Response.ok().entity(producerTemplate.requestBody("direct:" + scheme, statement, String.class));
        } catch (Exception e) {
                return javax.ws.rs.core.Response.serverError().entity(e.getMessage());
        }
            
    }
```

Comment (svkcemk) [reply integration-tests/validator/src/test/java/org/apache/camel/quarkus/component/validator/it/ValidatorTest.java]: > The tests are failing. You might want to forward the exception message to the client in `ValidatorResource`. Something like
> 
> ```
>     public javax.ws.rs.core.Response processOrderInXml(String statement, @PathParam("scheme") String scheme) {
>         try {
>             return javax.ws.rs.core.Response.ok().entity(producerTemplate.requestBody("direct:" + scheme, statement, String.class));
>         } catch (Exception e) {
>                 return javax.ws.rs.core.Response.serverError().entity(e.getMessage());
>         }
>             
>     }
> ```

**Review by ppalaga — APPROVED** [review]

**Comment by ppalaga** [comment]
Thanks @svkcemk !

**Pull request merged**
